packages/account-tree-controller/src/AccountTreeController.ts

```diff
@@ -394,30 +397,38 @@ export class AccountTreeController extends BaseController<
     wallet: AccountWalletObject,
     group: AccountGroupObject,
   ) {
-    const persistedMetadata = this.state.accountGroupsMetadata[group.id];
+    const persistedGroupMetadata = this.state.accountGroupsMetadata[group.id];
 
     // Apply persisted name if available (including empty strings)
-    if (persistedMetadata?.name !== undefined) {
-      group.metadata.name = persistedMetadata.name.value;
+    if (persistedGroupMetadata?.name !== undefined) {
+      group.metadata.name = persistedGroupMetadata.name.value;
     } else if (!group.metadata.name) {
       // Get the appropriate rule for this wallet type
       const rule = this.#getRuleForWallet(wallet);
       const typedWallet = wallet as AccountWalletObjectOf<typeof wallet.type>;
-      const typedGroup = typedWallet.groups[group.id] as AccountGroupObject;
-
-      // Calculate group index based on position within sorted group IDs
-      // We sort to ensure consistent ordering across all wallet types:
-      // - Entropy: group IDs like "entropy:abc/0", "entropy:abc/1" sort to logical order
-      // - Snap/Keyring: group IDs like "keyring:ledger/0xABC" get consistent alphabetical order
-      const sortedGroupIds = Object.keys(wallet.groups).sort();
-      let groupIndex = sortedGroupIds.indexOf(group.id);
-
-      // Defensive fallback: if group.id is not found in sortedGroupIds (should never happen
-      // in normal operation since we iterate over wallet.groups), use index 0 to prevent
-      // passing -1 to getDefaultAccountGroupName which would result in "Account 0"
-      /* istanbul ignore next */
-      if (groupIndex === -1) {
+      const typedGroup = typedWallet.groups[group.id] as AccountGroupObjectOf<
+        typeof group.type
+      >;
+
+      // Calculate group index for default naming
+      let groupIndex: number;
+
+      // For entropy-based multichain groups, use the actual groupIndex from metadata
+      if (group.type === AccountGroupType.MultichainAccount) {
+        groupIndex = group.metadata.entropy.groupIndex;
+      } else {
+        // For other wallet types (snap/keyring), use per-wallet sequential numbering
+        // to avoid the alphabetical sorting issue that can cause duplicate names
+        const existingGroupNames = Object.values(wallet.groups)
+          .map((g) => g.metadata.name)
+          .filter((name) => /^Account \d+$/u.test(name))
+          .map((name) => parseInt(name.replace('Account ', ''), 10) - 1); // Convert to 0-based
+
+        // Find the first available index
         groupIndex = 0;
+        while (existingGroupNames.includes(groupIndex)) {
+          groupIndex += 1;
+        }
       }
 
       // Use computed name first, then fallback to default naming if empty
```

Comment:
Below this part, we could still have issue with multichain account names IMO:
- You have your 1st multichain account -> "Account 1"
- You rename it to "Account 2"
- You create your 2nd multichain account -> "Account 2"

We now have 2 accounts named "Account 2".

We could maybe run a small conflict logic that if your default/computed name already exists, then we add a prefix (until we find a unique name).

WDYT?